src/api/v2/WorkspaceResourceProviderManager.ts

```diff
@@ -0,0 +1,23 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { ProvideResourceOptions, WorkspaceResource, WorkspaceResourceProvider } from "./v2AzureResourcesApi";
+import { ResourceProviderManagerBase } from './ResourceproviderManagerBase';
+import { isArray } from '../../utils/v2/isArray';
+
+export class WorkspaceResourceProviderManager extends ResourceProviderManagerBase<WorkspaceResource, WorkspaceResourceProvider> {
+    constructor(extensionActivator: () => Promise<void>) {
+        super(extensionActivator);
+    }
+
+    async provideResources(folder: vscode.WorkspaceFolder, options?: ProvideResourceOptions): Promise<WorkspaceResource[]> {
+        const resourceProviders = await this.getResourceProviders();
+
+        const resources = await Promise.all(resourceProviders.map(resourceProvider => resourceProvider.provideResources(folder, options)));
```

Comment:
Noticed we could be more consistent with the naming. We have `provideResources` on `WorkspaceResourceProvider` and `getResources` on `ApplicationResourceProvider`.

Reply:
Good call.  The `get` verb seems to be more prevalent in VS Code, so I'll switch to that.  Updated.